File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

vector<int> solution(int n, vector<string> words) {
    vector<int> answer;
    
            vector<string> prev;

        vector<vector<int>> cnt(n + 1, vector<int>(1, 0));

        int end = 0;
        for (int i = 0; i < words.size(); i++)
        {
            cnt[i%n+1][0] += 1;

            if (find(prev.begin(), prev.end(), words[i]) != prev.end())
            {
                end = i%n + 1;
                break;
            }
            else if (!prev.empty() && prev.back().back() != words[i].front())
            {
                end = i % n + 1;
                break;
            }
            else prev.push_back(words[i]);
        }
        answer.push_back(end);
        answer.push_back(cnt[end].back());

    return answer;
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.cs
using System;
using System.Linq;
using System.Collections.Generic;
class Solution
{
    public int[] solution(int n, string[] words)
    {
        int[] answer = new int[2];
        List<string> res = new List<string>();
        for(int i = 0; i<words.Length;i++)
        {
            
            if(res.Count != 0)
            {
                if(res[res.Count-1].Last() != words[i][0])
            {
                answer[0] = (i%n)+1;
                answer[1] = i/n+1;
                break;
            }
            }
            
            
            if(!res.Contains(words[i]))
            {
                 res.Add(words[i]);
            }
            else
            {
                answer[0] = (i%n)+1;
                answer[1] = i/n+1;
                break;
            }
        }
        
        return answer;
    }
}